keyboards/feker/ik75/keymaps/bkzshen/keymap.c

```diff
@@ -0,0 +1,256 @@
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    }
+                    break;
+                    case (LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR): {
+                        rgb_matrix_set_flags(LED_FLAG_UNDERGLOW);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    }
+                    break;
+                    case (LED_FLAG_UNDERGLOW): {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    }
+                    break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                        rgb_matrix_enable_noeeprom();
+                    }
+                    break;
+                }
+            }
+            return false;
+	}
+
+    return true;
+}
+
+void rgb_matrix_indicators_user(void) {
+    rgb_matrix_set_color(46, 0, 0, 0);
+    rgb_matrix_set_color(104, 0, 0, 0);
+
+    uint8_t red = host_keyboard_led_state().caps_lock ? 255 : 0;
+    uint8_t green = host_keyboard_led_state().scroll_lock ? 255 : 0;
+    uint8_t blue = keymap_config.no_gui ? 255 : 0;
+
+
+    if ((rgb_matrix_get_flags() & LED_FLAG_KEYLIGHT)) {
+        if (host_keyboard_led_state().num_lock) {
+            rgb_matrix_set_color(46, 255, 0, 0);
+        }
+        rgb_matrix_set_color(104, red, green, blue);
+    } else {
+        if (host_keyboard_led_state().num_lock) {
+            rgb_matrix_set_color(46, 255, 0, 0);
+        } else {
+            rgb_matrix_set_color(46, 0, 0, 0);
+        }
+        rgb_matrix_set_color(104, red, green, blue);
+    }
+}
+
+#ifdef ENCODER_ENABLE
+
+#define ENCODERS 3
+static uint8_t  encoder_state[ENCODERS] = {0};
+static keypos_t encoder_cw[ENCODERS]    = {{3, 5}};
+static keypos_t encoder_ccw[ENCODERS]   = {{4, 5}};
+
+void encoder_action_unregister(void) {
+    for (int index = 0; index < ENCODERS; ++index) {
+        if (encoder_state[index]) {
+            keyevent_t encoder_event = (keyevent_t) {
+                .key = encoder_state[index] >> 1 ? encoder_cw[index] : encoder_ccw[index],
+                .pressed = false,
+                .time = (timer_read() | 1)
+            };
+            encoder_state[index] = 0;
+            action_exec(encoder_event);
+        }
+    }
+}
+
+void encoder_action_register(uint8_t index, bool clockwise) {
+    keyevent_t encoder_event = (keyevent_t) {
+        .key = clockwise ? encoder_cw[index] : encoder_ccw[index],
+        .pressed = true,
+        .time = (timer_read() | 1)
+    };
+    encoder_state[index] = (clockwise ^ 1) | (clockwise << 1);
+    action_exec(encoder_event);
+}
+
+void matrix_scan_user(void) {
+    encoder_action_unregister();
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    encoder_action_register(index, clockwise);
+	return false;
+}
+#endif
```

Comment:
Should probably use encoder maps, nowadays.

Reply:
Ideally, this should be using the Encoder Map feature, and VIA actively supports it now.

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

Reply:
I've just pushed new commit, that using encoder map for VIA keymap now. Thanks!